### DISCUSSION TITLE
>2000 classes in Pimcore

### DISCUSSION THREAD

**Discussion by samikadze** [pimcore/pimcore#15665] [Q&A / Support]
Hello,

question for those, who have experience with creating and managing >2000 classes in Pimcore.

Have you done anything to speed up Pimcore, or did everything work fine?

For now I have to wait at least 30 seconds every time I save any class after editing.

Thanks!

**Comment by johanlindgren** [comment]
2000 is really many classes ...

There is an official recommendation on using Pimcore Classification Stores rather than object bricks when the number of different bricks for the same usecase are more than 30 (https://pimcore.com/docs/platform/Pimcore/Best_Practice/Object_Bricks_vs_Classification_Store/)

It's also possible to model attribute objects and have objects related for each atribute / attribute group and thsi way moving the growing quantity of attribiutes into the datamodel itself.

If you can share some details on your usecase it's probably easier to comment on the best use in your case

**Reply by samikadze** [reply]
Thanks for your reply. We are already thinking of switching to Classification Stores. But I suppose this will not help us significantly. 

Here is the reason why we need 2000 classes.
1. We are going to have approx. 100 product types (categories), and create a class for each product type.
2. In each of those 100 classes we are going to have 15-20 attributes with "Many-to-Many Object Relation". And for each of those relations we'll create a new class. The reason for using many-to-one object realtions as attributes (but not "Select", for example) is that we need to store some additional data for each of values, not just its name or value.

**Reply by dpfaffenbauer** [reply]
@samikadze Wouldn't it be better to have 1 Product Class with Object Bricks or Classification Store? Same for the attribute classes. 2k classes sounds like to much and there for sure is a better way

**Reply by samikadze** [reply]
Yes, we can have 1 Product class with Classification Store or Object bricks, but this will decrease amount of classes from 2000 to 1901. 

Other 1901 classes we still need to create "Many-to-Many Object Relation" attributes for products. As I've said, the reason for using many-to-one object realtions as attributes (but not "Select", for example) is that we need to store some additional data for each attribute value, not just its name/title.
